receiver/statsdreceiver/transport/server_test.go

```diff
@@ -16,89 +16,139 @@ import (
 	"go.opentelemetry.io/collector/consumer/consumertest"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/common/testutil"
-	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/protocol"
+	statsdtestutil "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/testutil"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/transport/client"
 )
 
+type testCase struct {
+	name              string
+	buildServerFn     func(addr string) (Server, error)
+	getFreeEndpointFn func(t testing.TB, network string) string
+	buildClientFn     func(network string, address string) (*client.StatsD, error)
+	testSkip          func(t testing.TB) bool
+}
+
 func Test_Server_ListenAndServe(t *testing.T) {
-	tests := []struct {
-		name          string
-		buildServerFn func(addr string) (Server, error)
-		buildClientFn func(host string, port int) (*client.StatsD, error)
-	}{
+	tests := []testCase{
+		{
+			name:              "udp",
+			getFreeEndpointFn: testutil.GetAvailableLocalNetworkAddress,
+			buildServerFn:     NewUDPServer,
+			buildClientFn:     client.NewStatsD,
+		},
 		{
-			name:          "udp",
-			buildServerFn: NewUDPServer,
-			buildClientFn: func(host string, port int) (*client.StatsD, error) {
-				return client.NewStatsD(client.UDP, host, port)
+			name:              "unix",
+			getFreeEndpointFn: statsdtestutil.CreateTemporarySocket,
+			buildServerFn:     NewUnixServer,
+			buildClientFn:     client.NewStatsD,
+			testSkip: func(t testing.TB) bool {
```

Comment:
nit: perhaps just using a `bool` for this variable `runtime.GOOS != "linux"`

Reply:
Done.